transformer_engine/common/fused_attn/fused_attn_f16_max512_seqlen.h

```diff
@@ -19,7 +19,7 @@
 
 namespace transformer_engine {
 #if (CUDNN_VERSION >= 8901)
-void fused_attn_max_512_fwd_qkvpacked(size_t batch, size_t max_seqlen, size_t num_head,
+void fused_attn_max_512_fwd_qkvpacked(size_t batch, size_t num_head, size_t max_seqlen,
```

Comment:
I'm curious why this PR changes the order of `H` and `S`, I think most of the implementation uses `B, S, H, D` as the layouts. So placing `S` before `H` seems more matching the most of the use cases. Like [NVTE_QKV_Layout](https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/common/include/transformer_engine/fused_attn.h#L33), [jax dot_product_attention](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/_autosummary/flax.linen.dot_product_attention.html), [dao-AILab flash attention](https://github.com/Dao-AILab/flash-attention#how-to-use-flashattention)